gammapy/estimators/points/sed.py

```diff
@@ -136,13 +136,31 @@ def run(self, datasets):
         if not datasets.energy_axes_are_aligned:
             raise ValueError("All datasets must have aligned energy axes.")
 
-        if "TELESCOP" in datasets.meta_table.colnames:
-            telescopes = datasets.meta_table["TELESCOP"]
-            if not len(np.unique(telescopes)) == 1:
-                raise ValueError(
-                    "All datasets must use the same value of the"
-                    " 'TELESCOP' meta keyword."
-                )
+        if (
```

Comment:
It does not seem necessary to test for all these conditions here. 
You only need to loop over all datasets in the `Datasets`, build the set of "TELESCOP" values and make sure it contains one element only.

Reply:
I found issues without the very specific statements. In particular, `dataset.meta_table` was sometimes None which means `dataset.meta_table.colnames` raises a TypeError